tests/timer/test_Event.cpp

```diff
@@ -0,0 +1,68 @@
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/timer/Event.hpp"
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * Events in time
+ * --------------
+ *
+ * This group of tests check the behavior of the events in time defined in @ref Event.hpp.
+ */
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+namespace Kokkos::utils::tests::timer
+{
+
+using namespace Kokkos::utils::timer;
+
+using event_t = Event<execution_space>;
+
+//! @test Check that @ref event_t satisfies the concept @ref Kokkos::utils::timer::IsEvent.
+TEST(Event, is_event)
+{
+    static_assert(IsEvent<event_t, execution_space, microseconds>);
+    static_assert(IsEvent<event_t, execution_space, milliseconds>);
+    static_assert(IsEvent<event_t, execution_space, seconds>);
+}
+
+//! @test Check the type of the event used by the primary template and the device specializations.
+TEST(Event, impl_event_type)
+{
+    using expt_impl_event_t =
+#if defined(KOKKOS_ENABLE_CUDA)
+    cudaEvent_t;
+#elif defined(KOKKOS_ENABLE_HIP)
+    hipEvent_t;
+#elif defined(KOKKOS_ENABLE_OPENMP)
+    std::chrono::steady_clock::time_point;
+#else
+    #error "Default execution space must be Cuda, HIP or OpenMP."
+#endif
+
+    static_assert(std::same_as<typename event_t::impl_event_t, expt_impl_event_t>);
+}
+
+//! @test Check the traits of the helper types @ref Kokkos::utils::timer::milliseconds and @ref Kokkos::utils::timer::seconds.
+TEST(Event, durations)
+{
+    static_assert(std::same_as<typename std::chrono::microseconds::period, std::ratio<1, 1000000>>);
+    static_assert(std::same_as<typename              microseconds::period, std::ratio<1, 1000000>>);
+    static_assert(std::same_as<typename              microseconds::rep,    double>);
+
+    static_assert(std::same_as<typename std::chrono::milliseconds::period, std::ratio<1, 1000>>);
+    static_assert(std::same_as<typename              milliseconds::period, std::ratio<1, 1000>>);
+    static_assert(std::same_as<typename              milliseconds::rep,    double>);
+
+    static_assert(std::same_as<typename std::chrono::seconds::period, std::ratio<1, 1>>);
+    static_assert(std::same_as<typename              seconds::period, std::ratio<1, 1>>);
+    static_assert(std::same_as<typename              seconds::rep,    double>);
+}
+
```

Comment:
You're missing a basic test along these lines:
```suggestion
TEST(Event, duration)
{
    Event begin, end;

    ASSERT_LT(end.duration<milli>(end), 0);
}
```